TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Infrastructure/Filters/CheckPersonExistsFilter.cs

```diff
@@ -7,20 +7,57 @@ namespace TeachingRecordSystem.SupportUi.Infrastructure.Filters;
 
 public class CheckPersonExistsFilter : IAsyncResourceFilter, IOrderedFilter
 {
+    private readonly bool _requireQts;
+
     public int Order => -200;
 
+    public CheckPersonExistsFilter(bool requireQts = false)
+    {
+        _requireQts = requireQts;
+    }
+
     public async Task OnResourceExecutionAsync(ResourceExecutingContext context, ResourceExecutionDelegate next)
     {
         var personId = context.RouteData.Values["personId"] as string;
+        if (personId is null)
+        {
+            var personIdQueryParameters = context.HttpContext.Request.Query["personId"];
+            if (personIdQueryParameters.Count == 1)
+            {
+                personId = personIdQueryParameters[0];
+            }
+        }
```

Comment:
Seems like we should throw here if we can't figure out the `personId`:
```suggestion
        var personId = context.RouteData.Values["personId"] as string ??
            context.HttpContext.Request.Query["personId"] ??
            throw new InvalidOperationException("Cannot extract personId from request.");
```